### ISSUE TITLE
Cypress test runner hangs immediately on running cypress open

### ISSUE THREAD

**Issue by jenkshields** [cypress-io/cypress#6378]
### Current behavior:

When running `yarn run cypress open`, the Cypress Test Runner opens but immediately hangs. There is no output in the terminal, unless I force quit Cypress, when it logs `The Test Runner unexpectedly exited via a exit event with signal SIGKILL` (which is to be expected).

Manually opening the globally installed app works fine.

Note! I ran into #6377 when installing, and worked around by installing `rjxs` and `rxjs-compat` - unsure if that could be related.

### Desired behavior:

Cypress Test Runner should open normally.

### Test code to reproduce

No tests are written - I've tried this both in a project without a cypress folder, and in one which has the folder filled with test examples with the same result.

### Versions

Cypress 4.0.1
Mac OS Catalina 10.15.3 Beta (19D62e)
Firefox Nightly 74.0a1
yarn v2

**Comment by thenakedthunder** [comment]
Hi, same happened to me. For me the problem arouse when I changed my build routine from 'npm run build' to 'yarn build'. I kinda think the problem is the location of the build output, it changed, or the cypress files cannot be found etc. It does not work with 'npx cypress open' either. 

F**k these package managers man, I sometimes feel I spend more time figuring out what went wrong again with my packages than with coding... So much of making your life easier.

**Comment by DanAtSpark** [comment]
I would like to update this issue with some details.
Firstly, Cypress hangs only on `yarn cypress open` or `yarn cypress version`
After a couple of minutes, cypress window (when running `yarn cypress open`) is starting but it doesn't have any content, just grey background.
I've tried to run cypress with the command
`$ DEBUG=cypress:* yarn cypress open`
And here's the output
```
cypress:cli:cli cli starts with arguments ["/usr/local/bin/node","$HOME/projects/cypressYarn2/.yarn/unplugged/cypress-npm-4.0.2-9b440563c1/node_modules/cypress/bin/cypress","open"] +0ms
  cypress:cli NODE_OPTIONS=--require $HOME/projects/cypressYarn2/.pnp.js +0ms
  cypress:cli:cli program parsing arguments +1ms
  cypress:cli:cli opening Cypress +1ms
  cypress:cli parsed cli options {} +169ms
  cypress:cli opening from options {"project":"$HOME/projects/cypressYarn2"} +0ms
  cypress:cli command line arguments ["--project","$HOME/projects/cypressYarn2"] +0ms
  cypress:cli verifying Cypress app +0ms
  cypress:cli checking environment variables +0ms
  cypress:cli checking if executable exists $HOME/.cache/Cypress/4.0.2/Cypress/Cypress +2ms
  cypress:cli Binary is executable? : true +3ms
  cypress:cli binaryDir is  $HOME/.cache/Cypress/4.0.2/Cypress +0ms
  cypress:cli Reading binary package.json from: $HOME/.cache/Cypress/4.0.2/Cypress/resources/app/package.json +0ms
  cypress:cli Found binary version 4.0.2 installed in: $HOME/.cache/Cypress/4.0.2/Cypress +2ms
  cypress:cli { verified: true } +6ms
  cypress:cli is Verified ? true +5ms
  cypress:cli DISPLAY environment variable is set to :1 on Linux
  cypress:cli Assuming this DISPLAY points at working X11 server,
  cypress:cli Cypress will not spawn own Xvfb
  cypress:cli 
  cypress:cli NOTE: if the X11 server is NOT working, Cypress will exit without explanation,
  cypress:cli   see https://github.com/cypress-io/cypress/issues/4034
  cypress:cli Solution: Unset the DISPLAY variable and try again:
  cypress:cli   DISPLAY= npx cypress run ... +0ms
  cypress:cli needs to start own Xvfb? false +0ms
  cypress:cli spawning, should retry on display problem? true +1ms
  cypress:cli passing DISPLAY :1 +1ms
  cypress:cli spawning Cypress with executable: $HOME/.cache/Cypress/4.0.2/Cypress/Cypress +0ms
  cypress:cli spawn args [ '--no-sandbox', '--', '--project', '$HOME/projects/cypressYarn2', '--cwd', '$HOME/projects/cypressYarn2' ] { detached: false, stdio: 'inherit' } +1ms
  cypress:ts Running without ts-node hook in environment "production" +0ms
  cypress:server:appdata path: $HOME/.config/Cypress/cy/production/browsers +0ms
  cypress:server:cypress starting cypress with argv [ '$HOME/.cache/Cypress/4.0.2/Cypress/Cypress', '--no-sandbox', '--', '--project', '$HOME/projects/cypressYarn2', '--cwd', '$HOME/projects/cypressYarn2' ] +0ms
  cypress:server:args argv array: [ '$HOME/.cache/Cypress/4.0.2/Cypress/Cypress', '--no-sandbox', '--project', '$HOME/projects/cypressYarn2', '--cwd', '$HOME/projects/cypressYarn2' ] +0ms
  cypress:server:args argv parsed: { _: [ '$HOME/.cache/Cypress/4.0.2/Cypress/Cypress' ], sandbox: false, project: '$HOME/projects/cypressYarn2', cwd: '$HOME/projects/cypressYarn2', invokedFromCli: true } +2ms
  cypress:server:util:proxy found proxy environment variables {} +0ms
  cypress:server:args options { _: [ '$HOME/.cache/Cypress/4.0.2/Cypress/Cypress' ], sandbox: false, project: '$HOME/projects/cypressYarn2', cwd: '$HOME/projects/cypressYarn2', invokedFromCli: true, config: {} } +1ms
  cypress:server:args argv options: { _: [ '$HOME/.cache/Cypress/4.0.2/Cypress/Cypress' ], sandbox: false, project: '$HOME/projects/cypressYarn2', cwd: '$HOME/projects/cypressYarn2', invokedFromCli: true, config: {}, projectRoot: '$HOME/projects/cypressYarn2' } +0ms
  cypress:server:cypress from argv [ '$HOME/.cache/Cypress/4.0.2/Cypress/Cypress', '--no-sandbox', '--project', '$HOME/projects/cypressYarn2', '--cwd', '$HOME/projects/cypressYarn2' ] got options { _: [ '$HOME/.cache/Cypress/4.0.2/Cypress/Cypress' ], sandbox: false, project: '$HOME/projects/cypressYarn2', cwd: '$HOME/projects/cypressYarn2', invokedFromCli: true, config: {}, projectRoot: '$HOME/projects/cypressYarn2' } +4ms
  cypress:server:appdata path: $HOME/.config/Cypress/cy/production +259ms
  cypress:server:cypress starting in mode interactive with options { _: [ '$HOME/.cache/Cypress/4.0.2/Cypress/Cypress' ], sandbox: false, project: '$HOME/projects/cypressYarn2', cwd: '$HOME/projects/cypressYarn2', invokedFromCli: true, config: {}, projectRoot: '$HOME/projects/cypressYarn2' } +67ms
  cypress:server:cypress running Electron currently +13ms
  cypress:server:appdata path: $HOME/.config/Cypress/cy/production +2s
  cypress:server:appdata path: $HOME/.config/Cypress/cy/production/cache +25s
  cypress:server:util:process_profiler current & mean memory and CPU usage by process group:
  cypress:server:util:process_profiler ┌─────────┬───────────────────┬──────────────┬────────────────┬────────────┬────────────────┬──────────┬──────────────┬─────────────┐
  cypress:server:util:process_profiler │ (index) │       group       │ processCount │      pids      │ cpuPercent │ meanCpuPercent │ memRssMb │ meanMemRssMb │ maxMemRssMb │
  cypress:server:util:process_profiler ├─────────┼───────────────────┼──────────────┼────────────────┼────────────┼────────────────┼──────────┼──────────────┼─────────────┤
  cypress:server:util:process_profiler │    0    │     'cypress'     │      1       │    '21843'     │     0      │       0        │  152.8   │    152.8     │    152.8    │
  cypress:server:util:process_profiler │    1    │ 'electron-shared' │      1       │    '21845'     │     0      │       0        │  38.99   │    38.99     │    38.99    │
  cypress:server:util:process_profiler │    2    │      'other'      │      2       │ '21856, 21857' │     0      │       0        │   3.96   │     3.96     │    3.96     │
  cypress:server:util:process_profiler │    3    │      'TOTAL'      │      4       │      '-'       │     0      │       0        │  195.75  │    195.75    │   195.75    │
  cypress:server:util:process_profiler └─────────┴───────────────────┴──────────────┴────────────────┴────────────┴────────────────┴──────────┴──────────────┴─────────────┘ +0ms
  cypress:server:saved_state making saved state from $HOME/.cache/Cypress/4.0.2/Cypress/resources/app/packages/server +0ms
  cypress:server:saved_state for project path $HOME/projects/cypressYarn2 +0ms
  cypress:server:saved_state state path for project $HOME/projects/cypressYarn2 +3ms
  cypress:server:appdata path: $HOME/.config/Cypress/cy/production/projects/cypressYarn2-80889c0a05e3db0ebcdb1822493b7cab/state.json +3m
  cypress:server:saved_state full state path $HOME/.config/Cypress/cy/production/projects/cypressYarn2-80889c0a05e3db0ebcdb1822493b7cab/state.json +0ms
  cypress:server:saved_state making new state file around $HOME/.config/Cypress/cy/production/projects/cypressYarn2-80889c0a05e3db0ebcdb1822493b7cab/state.json +1ms
  cypress:server:file get values from $HOME/.config/Cypress/cy/production/projects/cypressYarn2-80889c0a05e3db0ebcdb1822493b7cab/state.json +0ms
  cypress:server:file attempt to get lock on $HOME/.config/Cypress/cy/production/projects/cypressYarn2-80889c0a05e3db0ebcdb1822493b7cab/state.json +2ms
  cypress:server:file gettin lock succeeded or failed for $HOME/.config/Cypress/cy/production/projects/cypressYarn2-80889c0a05e3db0ebcdb1822493b7cab/state.json +10ms
  cypress:server:file read $HOME/.config/Cypress/cy/production/projects/cypressYarn2-80889c0a05e3db0ebcdb1822493b7cab/state.json +0ms
  cypress:server:file read succeeded or failed for $HOME/.config/Cypress/cy/production/projects/cypressYarn2-80889c0a05e3db0ebcdb1822493b7cab/state.json +4ms
  cypress:server:file attempt to unlock $HOME/.config/Cypress/cy/production/projects/cypressYarn2-80889c0a05e3db0ebcdb1822493b7cab/state.json +0ms
  cypress:server:file unlock succeeded or failed for $HOME/.config/Cypress/cy/production/projects/cypressYarn2-80889c0a05e3db0ebcdb1822493b7cab/state.json +1ms
  cypress:server:windows creating electron window with options { backgroundColor: '#dfe2e4', width: 800, height: 550, minWidth: 458, minHeight: 400, x: null, y: null, type: 'INDEX', devTools: false, trackState: { width: 'appWidth', height: 'appHeight', x: 'appX', y: 'appY', devTools: 'isAppDevToolsOpen' }, onBlur: [Function: onBlur], onFocus: [Function: onFocus], onClose: [Function: onClose], show: true, frame: true, transparent: false, icon: NativeImage {}, webPreferences: { preload: '$HOME/.cache/Cypress/4.0.2/Cypress/resources/app/packages/server/lib/ipc/ipc.js', partition: null, webSecurity: false, nodeIntegration: false, backgroundThrottling: false, backgroundColor: '#dfe2e4', transparent: false }, url: 'file://$HOME/.cache/Cypress/4.0.2/Cypress/resources/app/packages/desktop-gui/dist/index.html', contextMenu: false, recordFrameRate: null, onPaint: null, onCrashed: [Function: onCrashed], onNewWindow: [Function: onNewWindow] } +0ms
[21843:0218/134052.892814:INFO:CONSOLE(11588)] "failed to compile wasm module: RangeError: WebAssembly.Compile is disallowed on the main thread, if the buffer size is larger than 4KB. Use WebAssembly.compile, or compile on a worker thread.", source: $HOME/projects/cypressYarn2/.pnp.js (11588)
[21843:0218/134052.892931:INFO:CONSOLE(11597)] "Uncaught RangeError: WebAssembly.Compile is disallowed on the main thread, if the buffer size is larger than 4KB. Use WebAssembly.compile, or compile on a worker thread.", source: $HOME/projects/cypressYarn2/.pnp.js (11597)
  cypress:server:util:process_profiler current & mean memory and CPU usage by process group:
  cypress:server:util:process_profiler ┌─────────┬───────────────────┬──────────────┬───────────────────────┬────────────┬────────────────┬──────────┬──────────────┬─────────────┐
  cypress:server:util:process_profiler │ (index) │       group       │ processCount │         pids          │ cpuPercent │ meanCpuPercent │ memRssMb │ meanMemRssMb │ maxMemRssMb │
  cypress:server:util:process_profiler ├─────────┼───────────────────┼──────────────┼───────────────────────┼────────────┼────────────────┼──────────┼──────────────┼─────────────┤
  cypress:server:util:process_profiler │    0    │     'cypress'     │      1       │        '21843'        │    0.08    │      0.04      │  205.27  │    179.04    │   205.27    │
  cypress:server:util:process_profiler │    1    │ 'electron-shared' │      3       │ '21845, 21999, 22517' │     0      │       0        │  139.71  │    89.35     │   139.71    │
  cypress:server:util:process_profiler │    2    │      'other'      │      2       │    '22645, 22646'     │     0      │       0        │   3.89   │     3.93     │    3.96     │
  cypress:server:util:process_profiler │    3    │      'TOTAL'      │      6       │          '-'          │    0.08    │      0.04      │  348.88  │    272.31    │   348.88    │
  cypress:server:util:process_profiler └─────────┴───────────────────┴──────────────┴───────────────────────┴────────────┴────────────────┴──────────┴──────────────┴─────────────┘ +10s
```

**Comment by missing1984** [comment]
Running into the same issue with yarn v2. Highly suspect it's yarn v2 Plug'n'Play related.

**Comment by DanAtSpark** [comment]
It looks like more details and some workaround (which didn't work for me, though) were provided in another issue
https://github.com/cypress-io/cypress/issues/6377

**Comment by maxime1992** [comment]
This is happening to me as well and I'm completely stuck :warning:.
Cypress doesn't work locally at all, no matter how hard I try.

I'm **not** using yarn 2 so probably not coming from there:
```
yarn --version
1.21.1
```

Here's what I tried so far:

# Cypress 3.6.0

- `rm -rf node_modules && yarn` :arrow_right:  :red_circle: 
- `yarn run cypress cache clear && rm -rf node_modules && yarn` :arrow_right:  :red_circle: 


# Cypress 4.4.0
- `rm -rf node_modules && yarn` :arrow_right:  :red_circle: 
- `yarn run cypress cache clear && rm -rf node_modules && yarn` :arrow_right:  :red_circle: 

No matter what, I'm always stuck on that screen:

![image](https://user-images.githubusercontent.com/4950209/79450109-555e5680-7fdc-11ea-9689-04deeb6dad6f.png)

After some time, the only error that showed up in the console was that:

```
$ cypress open --project ./apps/public-site-e2e
It looks like this is your first time using Cypress: 4.4.0

  ✔  Verified Cypress! /home/maxime/.cache/Cypress/4.4.0/Cypress

Opening Cypress...
[545:0416/121013.504561:WARNING:x11_util.cc(1443)] X error received: serial 4770, error_code 3 (BadWindow (invalid Window parameter)), request_code 3, minor_code 0 (X_GetWindowAttributes)
[545:0416/121013.504660:WARNING:x11_util.cc(1443)] X error received: serial 4782, error_code 3 (BadWindow (invalid Window parameter)), request_code 3, minor_code 0 (X_GetWindowAttributes)
[545:0416/121013.504702:WARNING:x11_util.cc(1443)] X error received: serial 4794, error_code 3 (BadWindow (invalid Window parameter)), request_code 3, minor_code 0 (X_GetWindowAttributes)
```

I don't think that's relevant but sharing in case.

**Comment by DanielCambray** [comment]
Same issue here since yesterday. It was working fine until then. I don't know what to do...

**Comment by maxime1992** [comment]
@brian-mann sorry for the ping but it feels like quite a huge issue. We're unable to run cypress at all. I'm still blocked and unable to work on any tests. Something which is quite worrying too: I tried cypress on another Linux machine where it was working fine on few days ago and it's broken too :shrug:. On the contrary, my colleagues running it on Mac don't have any issue.

**Comment by DanielCambray** [comment]
Indeed, still not working here. I've tried many things, change node version, npm, yarn. Still the same issue.

I have managed to find an alternative, it's not perfect, far less stable, but at least I can run some tests. You can try to launch Cypress with docker. It works on ubuntu :

```
docker run -it \
  -v $PWD:/e2e \
  -v /tmp/.X11-unix:/tmp/.X11-unix \
  -v /var/run/dbus/system_bus_socket:/var/run/dbus/system_bus_socket \
  -w /e2e \
  -e DISPLAY=unix${DISPLAY} \
  --network=host \
  --entrypoint cypress \
  cypress/included:4.4.0 open --project .
```

**Comment by maxime1992** [comment]
So painful :( But thanks for sharing the solution @DanielCambray

**Comment by curvaton** [comment]
Hi,

I went through the same problem today. I have Ubuntu 18.04. On my side it appears it came from Chromium. I uninstalled it and now everything is back to normal. I found this solution remembering that this morning I had a notification that Chromium had been updated.

**Comment by maxime1992** [comment]
Oh my... :scream: @curvaton thanks! I didn't have to reinstall chromium-browser but just **remove it**.

What's going on... No idea! I still have chromium after "removing" it and now cypress does work :shrug: 

To be clear I just ran 
```
sudo apt remove --purge chromium-browser
```

**Comment by lujoamel4** [comment]
I tried all of these solutions. However, it didn't work until I have installed cypress@4.4.1.

**Comment by flotwig** [comment]
@maxime1992 it sounds like you're running into this bug: #7020 - try cypress@4.4.1

**Comment by jennifer-shehane** [comment]
Hey everyone, recently there was an issue where Cypress would hang during opening in Ubuntu due to their changing how Chromium is installed. This was a priority fix for us, which is why we released a patch fix ahead of schedule as [4.4.1](https://on.cypress.io/changelog#4-4-1). Please upgrade to this version.

These recent comments appear to be unrelated to the original issue that was opened - which involve `cypress open` not working in Yarn 2. Closing this issue with yarn 2 as duplicate of #6377
